src/Language/PureScript/CoreImp/AST.hs

```diff
@@ -92,6 +93,12 @@ data AST
   -- ^ instanceof check
   | Comment (Maybe SourceSpan) [Comment] AST
   -- ^ Commented JavaScript
+  | Pure (Maybe SourceSpan) AST
```

Comment:
Why add `Pure` to the CoreImp AST, instead of using a `Comment` node? We never match on it except to emit JavaScript, and it'll be one more constructor to have to handle when writing CoreImp optimization passes. Are there non-JS backends that share our CoreImp AST instead of working from CoreFn or something?

Reply:
I think at one point it wrapped a full expression with parens and all that. Now that it just inserts a comment we can probably remove it.

Reply:
> Are there non-JS backends that share our CoreImp AST instead of working from CoreFn or something?

I don't think there are.